File: 1510-find-lucky-integer-in-an-array/find-lucky-integer-in-an-array.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        int ans = 0;
        map<int, int> mp;

        for(int i = 0; i < arr.size(); i++) {
            mp[arr[i]]++;
        }

        for(auto x: mp) {
            if(x.first == x.second) ans = x.first;
        }
        return ans == 0 ? -1 : ans;
    }
};

File: 1510-find-lucky-integer-in-an-array/find-lucky-integer-in-an-array.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        ans = -1
        arr.sort()
        mp = defaultdict(int)
        for x in arr:
            mp[x] += 1
            

        for key, value in mp.items():
            if key == value : ans = key

        return ans